### PR TITLE
Icon settings for Plone 5

### PR DESCRIPTION
**Pull request by espenmn**
Fixed icon vocabulary for plone 5. Also fixed conditions in template and moved image before the title.
Probably: the settings should default to not showing image or description.

### CODE REVIEW

File: src/webcouturier/dropdownmenu/vocabularies.py [hunk 1]
```diff
@@ -1,31 +1,45 @@
...
+#from zope.schema.interfaces import IVocabularyFactory
 
 from webcouturier.dropdownmenu import msg_fact as _
 
-from zope.component.hooks import getSite
-
+try:
+    from zope.app.component.hooks import getSite
+except ImportError:
+    from zope.component.hooks import getSite
```
Comment (fulv) [inline src/webcouturier/dropdownmenu/vocabularies.py]: Shouldn't we first try to import from `zope.component.hooks`, and only do the bbb thing upon exception ?

Comment (espenmn) [inline src/webcouturier/dropdownmenu/vocabularies.py]: I have no opinion on this, I just used the approach I found in another (collective) product.
If you want, I can change the order

**Comment by jensens** [comment]
needs a rebase.

**Comment by jensens** [comment]
Please re-submit against master, thanks.

**Pull request closed without merging**